fenicsprecice/fenicsprecice.py

```diff
@@ -117,6 +117,7 @@ def create_coupling_expression(self):
         coupling_expression : Object of class dolfin.functions.expression.Expression
             Reference to object of class SegregatedRBFInterpolationExpression.
         """
+        assert(self._fenics_dims == 2), "Boundary conditions of Expression objects are only allowed for 2D cases"
```

Comment:
Does this mean that in 3D we currently can only use PointSources? Together with the restriction that parallel execution is not possible with PointSources, this is quite restrictive, because you can only do 3D runs (potentially big) in a sequential fashion.

Can you document this restriction in an issue?

Reply:
Yes we currently have the following restrictions:

* Cases using PointSource objects to modify boundary conditions can be run only in serial. 3D functionality is possible 
* Cases using Expressions objects to modify boundary conditions can be run in parallel but 2D

I'll open a separate issue about this

Reply:
Issue: https://github.com/precice/fenics-adapter/issues/149